### PR TITLE
Email fixes part2

### PR DESCRIPTION
**Pull request by HasnatL-Adapptlimited**
# Description

- Dropdown values filterable on DL page
- Can save DL selected emails from Select Filter and Select from List
- All emails selected from record in Select from List

## Tickets
- [x] https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/96348/ 
- [x] https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/92816/
- [x] https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/96347/
- [ ] https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/96350/
- [x] https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/96351/
- [x] https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/96343/

## Other Issues Fixed

- Tooltips added to header in the last dropdown
- Operator logic fix
- Warning message showing when DL dataset filter has no values

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by HasnatL-Adapptlimited** [comment]
Hey @AntoineRelief 

Opened PR for fixes

CC: @Vaishali1709 @morgang-adappt

File: libs/shared/src/lib/components/email/components/email-template/email-template.component.ts [hunk 1]
```diff
@@ -173,6 +169,10 @@ export class EmailTemplateComponent implements OnInit, OnDestroy {
 
   /** Show preview button for select with fiter option  */
   public showBtnPreview = false;
+  /**
+   * Subject to destroy
+   */
+  destroy$: Subject<boolean> = new Subject<boolean>();
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/components/email-template/email-template.component.ts:175]: @NehaSakariya-Adappt 
it's a good practice indeed, but we're reinventing the wheel there

there is already the unsubscribecomponent which does that

File: libs/shared/src/lib/components/email/components/email-template/email-template.component.html [hunk 2]
```diff
@@ -498,25 +497,32 @@
           </tr>
         </tbody>
       </table>
+      <div *ngIf="activeSegmentIndex === 2">
+        <label
+          class="block text-gray-700 text-sm"
+          *ngIf="dataList?.length === 0"
+        >
+          {{ 'components.select.distribution.noPreviewData' | translate }}
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/components/email-template/email-template.component.html:505]: @HasnatL-Adapptlimited 
I think we should reuse as much as possible the existing UI elements

I'll replace that
but in some many places, you are using the alert
There, we are introducing a new type of UI element, which breaks the overall feeling

File: libs/shared/src/lib/components/email/components/email-template/email-template.component.html [hunk 3]
```diff
@@ -498,25 +497,32 @@
...
+        {{ 'components.select.distribution.apply' | translate }}
       </ui-button>
     </div>
   </div>
 
   <div *ngIf="activeSegmentIndex === 2">
     <!-- Apply filter -->
     <ui-button
-      *ngIf="datasetFilterInfo.controls"
+      *ngIf="datasetFilterInfo.controls && showBtnPreview"
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/components/email-template/email-template.component.html:525]: @HasnatL-Adapptlimited 
better put that on the parent element, as the 2 children element buttons are using the same logic

**Comment by AntoineRelief** [comment]
@HasnatL-Adapptlimited @Vaishali1709 @morgang-adappt

merging the branch, I'll now deploy it on UAT

thus, it doesn't seem to fix this one:

https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/96350/

**Pull request merged**